internal/bundleutil/bundle.go

```diff
@@ -0,0 +1,39 @@
+package bundleutil
+
+import (
+	"encoding/json"
+	"fmt"
+
+	bsemver "github.com/blang/semver/v4"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	ocv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+)
+
+func GetVersion(b declcfg.Bundle) (*bsemver.Version, error) {
+	for _, p := range b.Properties {
+		if p.Type == property.TypePackage {
+			var pkg property.Package
+			if err := json.Unmarshal(p.Value, &pkg); err != nil {
+				return nil, fmt.Errorf("error unmarshalling package property: %w", err)
+			}
+			vers, err := bsemver.Parse(pkg.Version)
+			if err != nil {
+				return nil, err
+			}
+			return &vers, nil
+		}
+	}
+	return nil, fmt.Errorf("no package property found in bundle %q", b.Name)
+}
+
+// MetadataFor returns a BundleMetadata for the given bundle. If the provided bundle is nil,
```

Comment:
Is this still accurate? Would something like the following be more accurate?
```suggestion
// MetadataFor returns a BundleMetadata for the given bundle name and version. If the provided bundle version is nil,
```

Reply:
Addressed this in two ways:
1. Added `name and version`
2. Dropped the second sentence because I changed the signature to avoid nil-pointer panic possibilities.